### PR TITLE
Addl. testing in test.proto; deprecate XXX_ fields; support trailing comments

### PR DESCRIPTION
**Pull request by lrstanley**
- Upgrade cli commands in `pb/code_gen.sh` to latest protoc & protoc-gen-go arguments.
- Add backwards compatible fields into `pb/test.proto` for testing legacy `XXX_` field skip functionality.
- Add deprecation notice for `XXX_` field functionality.
- Spruce up README.
- Add support for trailing comments. This is a trivial addition, but it may be able to help clean up some folks proto files. Now both standard comment formats can be used. Example:
```proto
// file: test.proto
syntax = "proto3";

package pb;
option go_package = "/pb";

message IP {
  string Address = 1; // @inject_tag: valid:"ip"
  string MAC = 2;     // @inject_tag: validate:"omitempty"
}
```

### CODE REVIEW

**Comment by lrstanley** [comment]
Looks as though the previous `XXX_` fields were removed in protoc v3.x.x. It looks as though they removed unknown fields completely for a period of time, however in 3.5 and up (I'm testing against protoc v3.6.1) [they re-added them](https://developers.google.com/protocol-buffers/docs/proto3#unknowns) -- however they do not follow the `XXX_` syntax, and don't look to be exported. See example:

https://github.com/favadi/protoc-go-inject-tag/blob/1662467b944e38b53461a33db2c07e7fda9b0e7d/pb/test.pb.go#L73-L83

I think if we want to continue to test `XXX_` fields, we'd have to have an older version of protoc, as well as a newer version to ensure both work. If we're fine not supporting older protoc implementations, we could remove the `XXX_` functionality altogether.

Let me know your thoughts.

**Comment by lrstanley** [comment]
Actually, I misunderstood where the `XXX_` fields came from. protoc-gen-go (the library, not the plugin, though this is still confusing...) was specifically generating them (it's possible protoc plays a part too, but I'm not familiar enough with the tool yet). Looking at:

https://github.com/golang/protobuf/issues/1311#issuecomment-824232405

> Also, the generated API guarantees unknown field preservation, does not guarantee exactly how that is implemented. Starting with v1.4 and above, we no longer generate a publicly visible `XXX_unrecognized` Go struct field, but rather generate an internal field. The same unknown field preservation behavior is still there though for marshaling and unmarshaling.

More [here](https://github.com/protocolbuffers/protobuf-go/releases/tag/v1.20.0#v1.20-generated-code):

> All `XXX` prefixed fields (e.g., `XXX_NoUnkeyedLiteral`, `XXX_unrecognized`, or `XXX_sizecache`) and methods (e.g., `XXX_Size`, `XXX_Marshal`, `XXX_Unmarshal`, `XXX_Merge`, `XXX_DiscardUnknown`, `XXX_InternalExtensions`, `XXX_OneofWrappers`, or `XXX_WellKnownType`) have been removed and are instead implemented using unexported fields in the generated struct for a protobuf message. Any code directly or indirectly depending on these fields or methods need to be migrated to use principled APIs that provide equivalent and stable functionality. The compatibility document for Go protobufs reserves the right to add, modify, or remove these declarations.

The short version: since v1.4.0 of `protoc-gen-go` (the tool), these fields are no longer exported (released Apr 13, 2020).

I.e. if we removed (or deprecated) the `XXX_` functionality, this would only be an issue for folks who haven't updated `protoc-gen-go` since April of 2020. Additionally, this would only be an issue if they pulled the latest version of this tool (tied with their old `protoc-gen-go` version). If so, they would need to roll back to an older version of `protoc-go-inject-tag`.

**Comment by favadi** [comment]
Thank @lrstanley for high quality contribution again.

> I.e. if we removed (or deprecated) the XXX_ functionality, this would only be an issue for folks who haven't updated protoc-gen-go since April of 2020.

April of 2020 is only like... one year ago, maybe we should just keep it for a little while longer.
